src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -147,21 +147,28 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 	private registerGroupListeners(group: IEditorGroupView): void {
 		const groupDisposables = new DisposableStore();
 
-		groupDisposables.add(group.onDidGroupChange(e => {
+		groupDisposables.add(group.onDidModelChange(e => {
 			switch (e.kind) {
 				case GroupChangeKind.EDITOR_ACTIVE:
 					if (group.activeEditor) {
 						this._onDidEditorsChange.fire([{ groupId: group.id, editor: group.activeEditor, kind: GroupChangeKind.EDITOR_ACTIVE }]);
 					}
-					this.handleActiveEditorChange(group);
-					this._onDidVisibleEditorsChange.fire();
 					break;
 				default:
 					this._onDidEditorsChange.fire([{ groupId: group.id, ...e }]);
 					break;
 			}
 		}));
 
+		// Need to separatly listen to the group change for things like active editor changing
+		// as this doesn't always change the model (This could be a bug that needs more investigation)
+		groupDisposables.add(group.onDidGroupChange(e => {
```

Comment:
This may be the only group event we handle specifically and I see it is being emitted in a few places after the model has been updated. So it may warrant for a follow up issue that we might need to have a specific UI event for when the editor has changed to emit this event to not change timing.

Reply:
Yeah it was surprising to me that the model wouldn't just be a drop in replacement for the UI. Why do we see editors activated caused by view's that don't propagate into the model. If anything I would suggest a debt follow up might be that the view call into the model for view changes and then any events are just emitted from the model since the model is the source of truth.

Reply:
Yeah I suggest we fold this issue into the other debt issue for merging the 2 events we have into 1.